(DOCS-3777) Update Pivotal Platform instructions (#14955)
DataDog/documentation: content/en/integrations/faq/pivotal_architecture.md

---
title: Datadog VMware Tanzu Application Service Integration Architecture
kind: faq
description: "Overview of the Datadog integration with VMware Tanzu Application Service"
further_reading:
- link: "https://www.datadoghq.com/blog/pivotal-cloud-foundry-architecture/"
  tag: "Blog"
  text: "VMware Tanzu Application Service architecture"
- link: "https://docs.datadoghq.com/integrations/pivotal_pks/"
  tag: "Documentation"
  text: Pivotal Container Service
---

## Overview

The following diagram provides an overview of the components of the Datadog integration with VMware Tanzu Application Service and the flow of data between them. The sections following provide further detail about the individual components and their interrelationships.

> {{< img src="integrations/pivotal/pivotal_datadog_diagram.png" alt=""  >}}

Author: Thanks for catching this!

### Firehose Nozzle

The Datadog Firehose Nozzle consumes information from your deployment’s Loggregator (PCF’s system for aggregating deployment metrics and application logs). The nozzle collects internal nozzle metrics, application metrics, organization metrics, and logs from the Firehose, and adds the corresponding tags and application metadata it collects from the CAPI. You can configure the metadata filter from the cluster monitoring tile for the Datadog Firehose Nozzle with an allow- and deny-list mechanism. Specify which metadata to add to the metrics collected from the nozzle, and view the metrics and their corresponding tags in the [Metrics Summary][5] and [Metrics Explorer][6].

#### Metadata Collected

   - Tags exposed from the application metadata following the autodiscovery tags format `tags.datadoghq.com/k=v`. These are the tags a user can add to the application metadata from the CAPI.

## Datadog Application Monitoring Tile Components:

### Buildpack

> The Datadog Buildpack installs the lightweight Datadog Container Agent and Datadog Trace Agent for APM inside the container alongside the application. The Agent is only launched or started if logs collection is enabled through setting `DD_LOGS_ENABLED=TRUE`, used to send application-level logs to Datadog. Otherwise, DogStatsD is launched to send metrics. When the application is running with the Datadog buildpack, you can pass multiple configuration options using environment variables for the application. The variables can come from the application manifest (`manifest.yml`) or the Cloud Foundry(CF) CLI using `cf set-env` command.

Reviewer: ```suggestion
The Datadog Buildpack installs the lightweight Datadog Container Agent and Datadog Trace Agent for APM inside the container alongside the application. The Agent is only launched or started if logs collection is enabled through setting `DD_LOGS_ENABLED=TRUE`, used to send application-level logs to Datadog. Otherwise, DogStatsD is launched to send metrics. When the application is running with the Datadog buildpack, you can pass multiple configuration options using environment variables for the application. The variables can come from the application manifest (`manifest.yml`) or the Cloud Foundry(CF) CLI using the `cf set-env` command.
```

#### Metadata Collected

>    - Tags extracted from the env var `VCAP_APPLICATION` (example, `application_id`, `name`, `instance_index`, `space_name`, `uris`), and the env var`CF_INSTANCE_IP` for the `cf_instance_ip` tag.

Reviewer: ```suggestion
   - Tags extracted from the `VCAP_APPLICATION` environment variable (for example, `application_id`, `name`, `instance_index`, `space_name`, `uris`), and the `CF_INSTANCE_IP` environment variable for the `cf_instance_ip` tag.
```

>    - Tags added using the environment variable `DD_TAGS`.

Reviewer: ```suggestion
   - Tags added using the `DD_TAGS` environment variable.
```

  text: "VMware Tanzu Application Service architecture"
- link: "https://docs.datadoghq.com/integrations/pivotal_pks/"
  tag: "Documentation"
  text: "Pivotal Container Service"
---

## Overview

This page describes the architecture behind the Datadog VMware Tanzu Application Service integration.

> {{< img src="integrations/pivotal/pivotal_datadog_diagram.png" alt=""An overview of the components of the Datadog integration with VMware Tanzu Application Service and the flow of data between them."  >}}

Reviewer: ```suggestion
{{< img src="integrations/pivotal/pivotal_datadog_diagram.png" alt="An overview of the components of the Datadog integration with VMware Tanzu Application Service and the flow of data between them."  >}}
```
